File: queryrepository/src/main/java/fau/datastreamMetric/models/Metric.java
package fau.datastreamMetric.models;
import jakarta.persistence.*;


@Inheritance(strategy = InheritanceType.JOINED)
@Entity
public class Metric {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private long id;

    protected String name;

    // km/h maybe?
    protected String unit;

    // maybe the metric has a special format
    protected String format;

    public Metric(String name, String unit,String format) {
        this.name = name;
        this.unit = unit;
        this.format = format;
    }

    public Metric() {

    }

    public String getName() {
        return name;
    }

    public String getUnit() {
        return unit;
    }

    public long getId() {
        return id;
    }
}

File: frontend-query-repository/src/app/models/metric.ts
/* tslint:disable */
/* eslint-disable */
import { Measurement } from '../models/measurement';
export interface Metric {

  /**
   * id of Metric
   */
  id?: number;
  mesurements?: Array<Measurement>;

  /**
   * Name of the Metric
   */
  name?: string;

  /**
   * Unit of the Metric
   */
  unit?: string;
}
